utils/plots.py

```diff
@@ -87,13 +87,13 @@ def feature_visualization(x, module_type, stage, n=32, save_dir=Path("runs/detec
     if ("Detect" not in module_type) and (
         "Segment" not in module_type
     ):  # 'Detect' for Object Detect task,'Segment' for Segment task
-        batch, channels, height, width = x.shape  # batch, channels, height, width
+        _batch, channels, height, width = x.shape  # batch, channels, height, width
         if height > 1 and width > 1:
             f = save_dir / f"stage{stage}_{module_type.split('.')[-1]}_features.png"  # filename
 
             blocks = torch.chunk(x[0].cpu(), channels, dim=0)  # select batch index 0, block by channels
             n = min(n, channels)  # number of plots
-            fig, ax = plt.subplots(math.ceil(n / 8), 8, tight_layout=True)  # 8 rows x n/8 cols
+            _fig, ax = plt.subplots(math.ceil(n / 8), 8, tight_layout=True)  # 8 rows x n/8 cols
```

Comment:
❗ **CRITICAL**: `fig` is still used later in `feature_visualization` (e.g., `fig.savefig(...)`), so renaming the variable here to `_fig` makes `fig` undefined at runtime and will raise a `NameError`. Please keep the original variable name.

**Suggested change:**
```suggestion
            fig, ax = plt.subplots(math.ceil(n / 8), 8, tight_layout=True)
```